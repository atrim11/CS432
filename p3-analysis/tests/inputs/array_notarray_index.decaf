def int main() {
    int x;
    x[0] = 10;  // Error: 'x' is not an array but is accessed with an index
}