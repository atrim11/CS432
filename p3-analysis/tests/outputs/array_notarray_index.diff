--- outputs/array_notarray_index.txt	2024-10-23 13:25:26.507708221 -0400
+++ expected/array_notarray_index.txt	2024-10-21 12:14:20.588884696 -0400
@@ -1,14 +1 @@
-Program [line 1]
-SYM TABLE:
- print_int : (int) -> void
- print_bool : (bool) -> void
- print_str : (str) -> void
- main : () -> int
-
-  FuncDecl name="main" return_type=int parameters={} [line 1]
-  SYM TABLE:
-
-    Block [line 1]
-    SYM TABLE:
-     x : int
-
+Non-array 'x' accessed as an array on line 3
