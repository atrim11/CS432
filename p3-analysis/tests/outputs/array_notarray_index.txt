Invalid: Variable 'x' on line 3 accessed as an array
