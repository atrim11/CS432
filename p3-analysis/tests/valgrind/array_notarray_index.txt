==576946== Memcheck, a memory error detector
==576946== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==576946== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==576946== Command: ../decaf inputs/array_notarray_index.decaf
==576946== 
Non-array 'x' accessed as an array on line 3
==576946== 
==576946== HEAP SUMMARY:
==576946==     in use at exit: 0 bytes in 0 blocks
==576946==   total heap usage: 1,800 allocs, 1,800 frees, 210,176 bytes allocated
==576946== 
==576946== All heap blocks were freed -- no leaks are possible
==576946== 
==576946== For lists of detected and suppressed errors, rerun with: -s
==576946== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
