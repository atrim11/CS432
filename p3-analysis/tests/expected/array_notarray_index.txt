Non-array 'x' accessed as an array on line 3
